contents/codeQuality/CodeQualityMetrics.md

```diff
@@ -0,0 +1,98 @@
+# Code Quality Metrics
+
+Authors: [Wilson Kurniawan](https://github.com/wkurniawan07)
+
+## Overview
+
+- It is undisputed that developers, code reviewers, testers, QA team members, and software architects alike want code to be as simple, understandable, and (put more positive adjectives here) as possible.
+- It is, however, disputable how "simple", "understandable", etc. is defined, because they are all qualitative. What is "..." for one person may not be so "..." for another; in fact, it may not be so for that person three months later.
+
+A quantitative approach needs to be adopted to measure those qualities precisely. **Code Quality Metrics** is the answer. It provides a **quantitative measure** of the quality of a piece of code.
+
+> Since quantitative measurements are essential in all sciences, there is a continuous effort by computer science practitioners and theoreticians to bring similar approaches to software development.
+> The goal is obtaining objective, reproducible and quantifiable measurements, which may have numerous valuable applications in schedule and budget planning, cost estimation, quality assurance testing, software debugging, software performance optimization, and optimal personnel task assignments.
+>
+> https://en.wikibooks.org/wiki/Introduction_to_Software_Engineering/Quality/Metrics
+
+In fact, [the Boeing 777 project attributed its success to excellent usage of Quality Metrics (both code and non-code)](https://thesai.org/Downloads/Volume3No1/Paper%2021-Survey%20on%20Impact%20of%20Software%20Metrics%20on%20Software%20Quality.pdf).
+
+*Note that the term "Code Quality" here is scoped to be the readability, understandability, and maintainability of the code. While performance can be considered a legitimate basis for some metrics (e.g. time taken to run certain operations, based on profiling), it is not considered for this purpose.*
+
+## Example of Metrics
+
+### Complexity Metrics
+
+Complexity metrics measures how "complex" methods, classes, packages, etc. are. It is arguably the most useful metric for the largest number of developer because it is the easiest to grasp, the most directly relevant to the coding activity, and applicable to most/all programming languages.
```

Comment:
Will be fixed. Complex here means difficult to understand and to maintain, which I also have added.